### PR TITLE
Anchor links

### PR DESCRIPTION
**Pull request by davidpmccormick**
Adding the option for a Prismic `page` to display a list of in-page anchors to sections in the body that are headed by `h2`s.

By default, this is toggled off in Prismic.

- [x] Update `prismic-model/page` once the newer boolean type (included in #5636) is merged.
- [x] Allow `InfoBlock` headings (`h2`s) to be picked up in the `OnThisPageAnchors` component
image

### CODE REVIEW

File: common/views/components/Body/Body.js [hunk 1]
```diff
@@ -35,19 +38,27 @@ const Map = dynamic(import('../Map/Map'), { ssr: false });
 
 type BodySlice = {|
   type: string,
-  weight: Weight,
+  weight?: Weight,
```
Comment (davidpmccormick) [inline common/views/components/Body/Body.js:41]: We currently have several body slices that don't have a weight property. Making it optional was the less involved fix.

File: common/model/pages.js [hunk 2]
```diff
@@ -1,9 +1,11 @@
 // @flow
 import type { GenericContentFields } from './generic-content-fields';
+import type { Link } from './link';
 
 export type Page = {|
   type: 'pages',
   ...GenericContentFields,
+  onThisPage: Link[],
```
Comment (gestchild) [inline common/model/pages.js:8]: didn't you add this to the GenericContentFields, which it probably isn't so maybe remove from there?

File: common/services/prismic/pages.js [hunk 3]
```diff
@@ -48,6 +48,7 @@ export function parsePage(document: PrismicDocument): Page {
   return {
     type: 'pages',
     ...genericFields,
+    showOnThisPage: data.showOnThisPage || false,
```
Comment (gestchild) [inline common/services/prismic/pages.js:51]: found this a bit confusing - my immediate question was," show what on this page? but with it now - and not sure I can think of a better name for OnThisPage

**Review by gestchild — APPROVED** [review]

File: common/model/pages.js (hunk 2 above)
Comment (davidpmccormick) [reply common/model/pages.js:8]: I did add it to the `GenericContentFields` and I didn't feel great about it. The reason I did is that they're derived from the parsing of the body, which itself is a generic field. I'll see about moving it out, though.

**Pull request merged**